File: backend/components/metiers/administration_publique/utils/views/conformity/conformity_views.test.js
// conformity_views.test.js – Tests unitaires conformité views threed (JS)
const { checkRGPD, checkAccessibility } = require('./conformity_views');
describe('Conformity Threed', () => {
  it('valide la conformité RGPD', () => {
    expect(checkRGPD({ nom: 'Test' }).conform).toBe(true);
    expect(checkRGPD({ password: '123' }).conform).toBe(false);
  });
  it('valide l’accessibilité', () => {
    expect(checkAccessibility({ lang: 'fr' }).accessible).toBe(true);
    expect(checkAccessibility({} ).accessible).toBe(false);
  });
});

File: backend/components/metiers/administration_publique/utils/views/conformity/conformity_views.test.py
"""
conformity_views.test.py – Tests unitaires conformité views threed (Python)
"""
from .conformity_views import check_rgpd, check_accessibility

def test_check_rgpd():
    assert check_rgpd({"nom": "Test"}) == (True, "Conforme RGPD")
    assert check_rgpd({"password": "123"})[0] is False

def test_check_accessibility():
    class View: lang = 'fr'
    assert check_accessibility(View()) == (True, "Accessible")
    class NoLang: pass
    assert check_accessibility(NoLang())[0] is False
